repos/fdbt-site/src/pages/globalSettingsPeriodValidity.tsx

```diff
@@ -0,0 +1,148 @@
+import RadioConditionalInput from '../components/RadioConditionalInput';
+import { isPeriodExpiry } from '../interfaces/typeGuards';
+import { getFareDayEnd } from '../data/auroradb';
+
+const title = 'Period Validity - Create Fares Data Service';
+const description = 'Period Validity selection page of the Create Fares Data Service';
+
+interface PeriodValidityProps {
+    errors?: ErrorInfo[];
+    fieldset: RadioConditionalInputFieldset;
+    csrfToken: string;
+    endOfFareDay?: string;
+}
+
+export const getFieldset = (errors: ErrorInfo[], endOfFareDay: string | undefined): RadioConditionalInputFieldset => {
+    const periodValidityFieldSet: RadioConditionalInputFieldset = {
+        heading: {
+            id: 'period-validity',
+            content: 'Is this ticket only valid on certain days or times?',
+            hidden: true,
+        },
+        radios: [
+            {
+                id: 'period-end-calendar',
+                name: 'periodValid',
+                value: 'endOfCalendarDay',
+                label: ' At the end of a calendar day',
+                radioButtonHint: {
+                    id: 'period-end-calendar-hint',
+                    content:
+                        'For example, a ticket purchased at 3pm would be valid until midnight on its day of expiry',
+                },
+            },
+            {
+                id: 'period-twenty-four-hours',
+                name: 'periodValid',
+                value: '24hr',
+                label: 'At the end of a 24 hour period from purchase',
+                radioButtonHint: {
+                    id: 'period-twenty-four-hours-hint',
+                    content: 'For example, a ticket purchased at 3pm will be valid until 3pm on its day of expiry',
+                },
+            },
+            {
+                id: 'period-end-of-service',
+                name: 'periodValid',
+                value: 'endOfServiceDay',
+                dataAriaControls: 'period-validity-end-of-service-required-conditional',
+                label: 'End of service day',
+                radioButtonHint: {
+                    id: 'period-end-of-service-hint',
+                    content:
+                        'For example, a ticket purchased at 3pm would be valid until the end of your service day on its day of expiry',
+                },
+                inputHint: {
+                    id: 'product-end-time-hint',
+                    content: 'Enter an end time for your service day',
+                    hidden: true,
+                },
+                inputType: 'text',
+                inputs: [
+                    {
+                        id: 'product-end-time',
+                        name: 'productEndTime',
+                        label: 'End time',
+                        disabled: true,
+                        defaultValue: errors.length > 0 && errors[0].userInput ? errors[0].userInput : endOfFareDay,
```

Comment:
Should this be changed so the `value` (rather than `default value`) is always endOfFareDay now?

Reply:
There is no `value` property on the `BaseReactElement`, I can add it if you want?